examples/ping.rs

```diff
@@ -0,0 +1,139 @@
+#[macro_use]
+extern crate log;
+extern crate env_logger;
+extern crate getopts;
+extern crate smoltcp;
+extern crate byteorder;
+
+mod utils;
+
+use std::str::{self, FromStr};
+use std::time::{Duration, Instant};
+use smoltcp::Error;
+use smoltcp::wire::{EthernetAddress, IpAddress, IpProtocol, Icmpv4Repr, Icmpv4Packet};
+use smoltcp::iface::{ArpCache, SliceArpCache, EthernetInterface};
+use smoltcp::socket::{AsSocket, SocketSet};
+use smoltcp::socket::{RawSocket, RawSocketBuffer, RawPacketBuffer};
+use std::collections::HashMap;
+use byteorder::{ByteOrder, NetworkEndian};
+
+const PING_INTERVAL_S: u64 = 1;
+const PING_TIMEOUT_S: u64 = 5;
+const PINGS_TO_SEND: usize = 4;
+
+fn main() {
+    utils::setup_logging();
+
+    let (device, args) = utils::setup_device(&["ADDRESS"]);
+    let remote_addr = IpAddress::from_str(&args[0]).expect("invalid address format");
+    let local_addr = IpAddress::v4(192, 168, 69, 2);
+
+    let raw_rx_buffer = RawSocketBuffer::new(vec![RawPacketBuffer::new(vec![0; 64])]);
+    let raw_tx_buffer = RawSocketBuffer::new(vec![RawPacketBuffer::new(vec![0; 128])]);
+    let raw_socket = RawSocket::new(raw_rx_buffer, raw_tx_buffer, local_addr, IpProtocol::Icmp);
+
+    let hardware_addr = EthernetAddress([0x02, 0x00, 0x00, 0x00, 0x00, 0x02]);
+    let arp_cache = SliceArpCache::new(vec![Default::default(); 8]);
+    let mut iface = EthernetInterface::new(
+        Box::new(device),
+        Box::new(arp_cache) as Box<ArpCache>,
+        hardware_addr,
+        [local_addr],
+    );
+
+    let mut sockets = SocketSet::new(vec![]);
+    let raw_handle = sockets.add(raw_socket);
+
+    let mut send_next = Duration::default();
+    let mut seq = 0;
+    let mut received = 0;
+    let mut echo_payload = [0xffu8; 40];
+    let mut waiting_queue = HashMap::new();
+
+    let startup_time = Instant::now();
+
+    loop {
+        {
+            let socket: &mut RawSocket = sockets.get_mut(raw_handle).as_socket();
+
+            let timestamp = Instant::now().duration_since(startup_time);
+            let timestamp_us = (timestamp.as_secs() * 1000000) +
+                (timestamp.subsec_nanos() / 1000) as u64;
+
+            if seq == PINGS_TO_SEND && waiting_queue.is_empty() {
+                break;
+            }
+
+            if socket.can_send() && seq < PINGS_TO_SEND && send_next <= timestamp {
+                NetworkEndian::write_u64(&mut echo_payload, timestamp_us);
+
+                let repr = Icmpv4Repr::EchoRequest {
+                    ident: 1,
+                    seq_no: seq as u16,
+                    data: &echo_payload,
+                };
+                let raw_payload = socket.send(repr.buffer_len(), remote_addr).expect(
+                    "Can't allocate packet",
+                );
+                let mut packet =
+                    Icmpv4Packet::new(raw_payload).expect("Can't create ICMPv4 packet");
```

Comment:
Let's just `unwrap()`. If we're going to add more proper error handling to examples it should be done in sync (and use `error!`, not `expect()`).